vue/test/components/dropdown/dropdown.test.js

```diff
@@ -0,0 +1,165 @@
+        const items = [
+            {
+                value: "text_1",
+                label: "Text 1"
+            },
+            {
+                value: "text_2"
+            }
+        ];
+        const component = base.getComponent("Dropdown", {
+            props: { items: items, selected: { 1: true } }
+        });
+
+        assert.strictEqual(component.vm.$data.selectedData[1], true);
+    });
+
+    it("should highlight item on item mouseover and dehighlight", async () => {
+        const items = [
+            {
+                value: "text_1",
+                label: "Text 1"
+            },
+            {
+                value: "text_2"
+            }
+        ];
+        const component = base.getComponent("Dropdown", {
+            props: { items: items }
+        });
+
+        const item1Index = 0;
+        const item1 = component.findAll(".dropdown-item").at(item1Index);
+        await item1.trigger("mouseenter");
+        await item1.trigger("mouseover");
+        assert.strictEqual(component.vm.$data.highlightedData[item1Index], true);
+        await item1.trigger("mouseleave");
+        assert.strictEqual(component.vm.$data.highlightedData[item1Index], undefined);
+
+        const item2Index = 1;
+        const item2 = component.findAll(".dropdown-item").at(item2Index);
+        await item2.trigger("mouseenter");
+        await item2.trigger("mouseover");
+        assert.strictEqual(component.vm.$data.highlightedData[item2Index], true);
+        await item2.trigger("mouseleave");
+        assert.strictEqual(component.vm.$data.highlightedData[item2Index], undefined);
+    });
+
+    it("should send the correct highlight event", async () => {
+        const items = [
+            {
+                value: "text_1",
+                label: "Text 1"
+            },
+            {
+                value: "text_2"
+            }
+        ];
+        const component = base.getComponent("Dropdown", {
+            props: { items: items }
+        });
+
+        const itemIndex1 = 0;
+        const item1 = component.findAll(".dropdown-item").at(itemIndex1);
+        await item1.trigger("mouseenter");
+        await item1.trigger("mouseover");
+        let highlightedItem = parseInt(
+            Object.keys(component.emitted("update:highlighted")[0][0])[0]
+        );
+        assert.strictEqual(component.emitted("update:highlighted").length, 1);
+        assert.strictEqual(highlightedItem, highlightedItem);
+        await item1.trigger("mouseleave");
+
+        const itemIndex2 = 1;
+        const item2 = component.findAll(".dropdown-item").at(itemIndex2);
+        await item2.trigger("mouseenter");
+        await item2.trigger("mouseover");
+        highlightedItem = parseInt(Object.keys(component.emitted("update:highlighted")[2][0])[0]);
+        assert.strictEqual(component.emitted("update:highlighted").length, 3);
+        assert.strictEqual(highlightedItem, itemIndex2);

```

Comment:
I think this test is too complex.
It would probably be better to make something like
```
component.vm.$on("update:highlighted", value => {
...
});
```
Where you listen for the event to be triggered, then do something like push the value to some array for example and then check if the values and order are what we are expecting